### ISSUE TITLE
BTCD syncing deprecation notice

### ISSUE THREAD

**Issue by matt24smith** [btcsuite/btcd#1520]
This is partially a continuation from [another issue thread](https://github.com/btcsuite/btcd/issues/1339), but I wanted to open a new issue to gain some attention to the problem and hopefully get a response from the main BTCD developers. 

First of all, I'm 100% on board with supporting alternative implementations of bitcoin nodes. Having multiple implementations interacting with the same blockchain is an essential part of a healthy blockchain ecosystem. Right now it appears that bitcoin core runs the vast majority of nodes in the network (judging solely based off logfiles describing peer connections to my node). This is all well and good, however if a critical flaw were to be found in bitcoin core, it would cripple most of the blockchain network and potentially impact trust and adoption of cryptocurrency as a whole. This is where alternative implementations such as BTCD play an important role: we need viable alternatives to turn to in the event that consensus decides the leading node implementation is flawed. 

BTCD very nearly works to solve this problem, but after some time testing, there exist some critical flaws preventing it from being seriously considered as an alternative node implementation. The syncing performance leaves much to be desired. Is it true that BTCD only syncs from a single peer to update the blockchain? This doesn't sound very decentralized. Additionally, the time to sync blocks increases exponentially as the block height increases, as you can see in the data collected from the logs below (the gap in the data was from where I stopped syncing to mess with configs). 

My fear is that the rate that new blocks are generated in the blockchain will exceed the rate at which they can be synced across BTCD nodes, which will render BTCD mostly useless in the near future if it can't keep up. I would really love to be able to recommend BTCD as a viable alternative node implementation, but until these problems are fixed, I can't recommend anybody to use it. 

![Figure_1](https://user-images.githubusercontent.com/8880061/72556099-531d1c80-3874-11ea-8e32-93adb7b88ea3.png)

**Comment by Roasbeef** [comment]
> My fear is that the rate that new blocks are generated in the blockchain will exceed the rate at which they can be synced across BTCD nodes

[Between this year and the last, `btcd `synced the chain in _less_ time, even accounting for chain size expansion](https://blog.lopp.net/bitcoin-node-performance-sync-tests/#btcd-v0-20-0-beta), which is no small feat. 

> he syncing performance leaves much to be desired. Is it true that BTCD only syncs from a single peer to update the blockchain? This doesn't sound very decentralized. 

That has virtually nothing to do with decentralization, it's a problem of network pipelining/throughput and utilization. 

If it's taking you weeks to sync the chain as is, you may want to upgrade to an SSD if you haven't already. I can bootstrap a new node in only a few days, and even less with the set of optimization PRs that are open in the repo. At the end of the day, this is an open source project, if you want to improve the status of things there're several open PRs you can help test if you aren't able to review or perform deeper analysis.

**Comment by matt24smith** [comment]
@Roasbeef thanks for your reply. I did suspect it to be a hardware I/O issue at first, but after profiling my disk use, it wasn't using the full throughput my hardware is capable of. Besides, is requiring an SSD to run the software really a solution? I ran my node in a docker container, so it remains possible that docker is the bottleneck. That being said, bitcoin core synced in a day on the same hardware in a similar docker configuration. Maybe I will look into the optimization PRs you mention and see if it makes a difference.
Your contributions to open source are much appreciated, I will definitely look into testing other open PRs if I have some spare time in the future.
